src/utils/callbackSet.ts

```diff
@@ -0,0 +1,38 @@
+///<reference path="../reference.ts" />
+
+module Plottable {
+  export module Utils {
+    export class CallbackSet<CB extends Function> {
+      private _values: CB[];
+
+      constructor() {
+        this._values = [];
+      }
+
+      public add(value: CB) {
+        if (this._values.indexOf(value) === -1) {
```

Comment:
`Set`s are usually known to work within O(1)... but this is clearly not the case.  Is it alright if we go against the convention for time complexities for `Set`s?